source/ptreferential/ptreferential_utils.cpp

```diff
@@ -291,30 +291,35 @@ type::Indexes get_indexes_from_code(const type::Type_e type,
 template <typename T>
 static
     typename boost::enable_if<typename boost::mpl::contains<nt::CodeContainer::SupportedTypes, T>::type, Indexes>::type
-    get_indexes_from_code_type(const std::string& key, const Data& data) {
+    get_indexes_from_code_type(const std::vector<std::string>& keys, const Data& data) {
     Indexes indexes;
     auto collection = data.pt_data->collection<T>();
-    for (const auto* obj : collection) {
-        auto codes = data.pt_data->codes.get_codes<T>(obj);
-        if (codes.find(key) == codes.end()) {
-            continue;
+    for (const auto& key : keys) {
+        for (const auto* obj : collection) {
```

Comment:
I did some test with 100000 stop_areas/stop_points looking for a stop_point at the end of the list. Surprisingly enough, there is no difference in term of performances. I have a similar order of magnitude of cache miss with the 2 versions when profiling with valgrind. 

Although, I agree (as per my first implementation) that keys are likely to be much smaller than the collection size. And therefore, this would me better this way.